syft/frameworks/torch/tensors/interpreters/paillier.py

```diff
@@ -0,0 +1,296 @@
+        response = hook_args.hook_response("__add__", response, wrap_type=type(self))
+        return response
+
+    def __sub__(self, *args, **kwargs):
+        """
+        Here is the version of the add method without the decorator: as you can see
+        it is much more complicated. However you misght need sometimes to specify
+        some particular behaviour: so here what to start from :)
+        """
+
+        if isinstance(args[0], th.Tensor):
+            data = self.child - args[0].numpy()
+            obj = PaillierTensor()
+            obj.child = data
+            return obj
+
+        if isinstance(self.child, th.Tensor):
+            self.child = self.child.numpy()
+
+        # Replace all syft tensor with their child attribute
+        new_self, new_args, new_kwargs = hook_args.unwrap_args_from_method(
+            "__sub__", self, args, kwargs
+        )
+
+        # Send it to the appropriate class and get the response
+        response = getattr(new_self, "__sub__")(*new_args, **new_kwargs)
+
+        # Put back SyftTensor on the tensors found in the response
+        response = hook_args.hook_response("__sub__", response, wrap_type=type(self))
+        return response
+
+    def __mul__(self, *args, **kwargs):
+        """
+        Here is the version of the add method without the decorator: as you can see
+        it is much more complicated. However you misght need sometimes to specify
+        some particular behaviour: so here what to start from :)
+        """
+
+        if isinstance(args[0], th.Tensor):
+            data = self.child * args[0].numpy()
+            obj = PaillierTensor()
+            obj.child = data
+            return obj
+
+        if isinstance(self.child, th.Tensor):
+            self.child = self.child.numpy()
+
+        # Replace all syft tensor with their child attribute
+        new_self, new_args, new_kwargs = hook_args.unwrap_args_from_method(
+            "__mul__", self, args, kwargs
+        )
+
+        # Send it to the appropriate class and get the response
+        response = getattr(new_self, "__mul__")(*new_args, **new_kwargs)
+
+        # Put back SyftTensor on the tensors found in the response
+        response = hook_args.hook_response("__mul__", response, wrap_type=type(self))
+        return response
+
+    def mm(self, *args, **kwargs):
+        """
+        Here is matrix multiplication between an encrypted and unencrypted tensor. Note that
+        we cannot matrix multiply two encrypted tensors because Paillier does not support
+        the multiplication of two encrypted values.
+        """
+        out = PaillierTensor()
+
+        # if self is not encrypted and args[0] is encrypted
+        if isinstance(self.child, th.Tensor):
+            out.child = self.child.numpy().dot(args[0].child)
+
+        # if self is encrypted and args[0] is not encrypted
+        else:
+            out.child = self.child.dot(args[0])
+
+        return out
```

Comment:
I think mm = __ matmul __ 
so you can add:
`__matmul__ == mm` to add support for `x @ y`